tests/integ/scala/test_datatype_suite.py

```diff
@@ -33,7 +36,69 @@
     VariantType,
     VectorType,
 )
-from tests.utils import Utils
+from tests.utils import IS_ICEBERG_SUPPORTED, IS_STRUCTURED_TYPES_SUPPORTED, Utils
+
+# Map of structured type enabled state to test params
+STRUCTURED_TYPES_EXAMPLES = {
+    True: pytest.param(
+        """
+            select
+              object_construct('k1', 1) :: map(varchar, int) as map,
+              object_construct('A', 'foo', 'B', 0.05) :: object(A varchar, B float) as obj,
+              [1.0, 3.1, 4.5] :: array(float) as arr
+            """,
+        [
+            ("MAP", "map<string(16777216),bigint>"),
+            ("OBJ", "struct<string(16777216),double>"),
+            ("ARR", "array<double>"),
+        ],
+        StructType(
+            [
+                StructField(
+                    "MAP",
+                    MapType(StringType(16777216), LongType(), structured=True),
+                    nullable=True,
+                ),
+                StructField(
+                    "OBJ",
+                    StructType(
+                        [
+                            StructField("A", StringType(16777216), nullable=True),
+                            StructField("B", DoubleType(), nullable=True),
+                        ],
+                        structured=True,
+                    ),
+                    nullable=True,
+                ),
+                StructField(
+                    "ARR", ArrayType(DoubleType(), structured=True), nullable=True
+                ),
+            ]
+        ),
+        id="structured-types-enabled",
+    ),
+    False: pytest.param(
```

Comment:
we finally will remove this example once all deployments supporting struct types?

Reply:
Yes, at some point in the future we will make a BCR that turns structured types on by default. That will probably be the best time to remove this.